### DISCUSSION TITLE
Pivot relation manager

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#10465] [Help] [answered]
How can I use pivot relation with relation manager? I searched for such a problem in the discussions, but I did not find what I needed.

```php
class OrderRoutesRelationManager extends RelationManager
{
    protected static string $relationship = 'orderRoutes';
}
```

Model
```php
class Order extends Model
{
    public function orderRoutes()
    {
        return $this->hasMany(OrderRoute::class, 'order_id');
    }
}
```

Pivot
```php
class OrderRoute extends Pivot
{
    protected $fillable = [
        'order_id',
        'route_id',
        'another_id',
    ];
}
```

Exception
`SQLSTATE[42S22]: Column not found: 1054 Unknown column 'order_route.id' in 'order clause'`

**order_route** table
`order_id` | `route_id` | `another_id`

**Comment by RoseRiyadh** [comment]
have you found the solution?

**Reply by ArtMin96** [reply]
@RoseRiyadh in my case I changed the relation
```php
public function orderRoutes()
{
    return $this->hasMany(OrderRoute::class, 'order_id');
}
```

**Comment by micraux** [comment, marked as answer]
If I got it right, you have 3 tables: `orders`, `routes` and `order_route` as the pivot table.

If it is the case, the relations in your `Order` and `Route` models should both be `BelongsToMany`:
[https://laravel.com/docs/10.x/eloquent-relationships#many-to-many-model-structure](https://laravel.com/docs/10.x/eloquent-relationships#many-to-many-model-structure)

**Reply by ArtMin96** [reply]
Thanks
